### ISSUE TITLE
Error when running DeterministicTraceTranslator

### ISSUE THREAD

**Issue by ztangent** [probcomp/Gen.jl#396]
When running `DeterministicTraceTranslator`, `run_first_pass` results in an error because it passes in `aux_trace` as `nothing` instead of an actual `Trace`, which prevents `FirstPassState` from being properly constructed:

```julia
ERROR: MethodError: no method matching get_choices(::Nothing)
Closest candidates are:
  get_choices(::var"##StaticIRTrace_line_model#460") at none:0
  get_choices(::var"##StaticIRTrace_line_step#441") at none:0
  get_choices(::Gen.CallAtTrace) at C:\Users\...\.julia\packages\Gen\k43lp\src\modeling_library\call_at\call_at.jl:52
  ...
Stacktrace:
 [1] Gen.FirstPassState(::Gen.DynamicDSLTrace{DynamicDSLFunction{Any}}, ::Nothing) at C:\Users\...\.julia\packages\Gen\k43lp\src\inference\trace_translators.jl:260
 [2] run_first_pass(::TraceTransformDSLProgram, ::Gen.DynamicDSLTrace{DynamicDSLFunction{Any}}, ::Nothing) 
at C:\Users\...\.julia\packages\Gen\k43lp\src\inference\trace_translators.jl:266
```

It looks like commit 77e5dab introduced a regression when it changed the internal structure of `FirstPassState` to be more stringent and catch certain kinds of errors.

**Comment by ztangent** [comment]
I'm now running into a separate bug, caused by a parsing error in the `@transform` macro: at line 129, `src_expr` should be `dest_expr` instead:

https://github.com/probcomp/Gen.jl/blob/ee903a4d206413c5379f90de1eb3aee52e15bb74/src/inference/trace_translators.jl#L128-L133

`@transform` also currently does not throw an error if `err` becomes true, which should be fixed.

**Comment by ztangent** [comment]
And one more bug: The following code assumes that `bij_state.aux_trace` is always a trace, but when using a `DeterministicTraceTranslator`, it is just `nothing`, which causes an error at line 144:

https://github.com/probcomp/Gen.jl/blob/ee903a4d206413c5379f90de1eb3aee52e15bb74/src/inference/trace_translators.jl#L139-L151

**Comment by ztangent** [comment]
Addressed by #397.
